### PR TITLE
Boost buffer content replace

### PR DESCRIPTION
**Pull request by mluts**
`:substitute` hangs my machine for 5-15 seconds on large files, this works much faster

### CODE REVIEW

File: plugin/cljfmt.vim [hunk 1]
```diff
@@ -70,7 +70,16 @@ function! s:GetFormattedFile()
...
     redir END
-    return s:FilterOutput(split(l:cljfmt_output, "\n"))
+    return s:FilterOutput(split(l:cljfmt_output, "\n"), 0)
+endfunction
+
+function! s:replaceBuffer(content) abort
+  let content = type(a:content) == v:t_list ? a:content : split(a:content, "\n")
+
+  if getline(1, '$') != content
+    %del
```
Comment (venantius) [inline plugin/cljfmt.vim:80]: What does this do?

**Comment by venantius** [comment]
Definitely appreciate the speedup. Would love to understand a bit more about what exactly the changes do since it's been a while since I wrote this. Mind breaking it down a bit more for me?

**Comment by mluts** [comment]
@venantius `:0,substitute/\_.*/\=formatted_output/g` works frustratingly slow, perhaps because whole file becomes matched using regexp `.*`.

`%del` is actually vim's `:[range]delete` command (see :help :delete)

`setline()`

                When {text} is a List then line {lnum} and following lines     
                will be set to the items in the list.  Example:                
                        :call setline(5, ['aaa', 'bbb', 'ccc'])                
                This is equivalent to:                                         
                        :for [n, l] in [[5, 'aaa'], [6, 'bbb'], [7, 'ccc']]    
                        :  call setline(n, l)                                  
                        :endfor

**Comment by venantius** [comment]
Great, thanks!

**Pull request merged**